### PR TITLE
Handle streaming failures correctly

### PR DESCRIPTION
**Pull request by GabrielBianconi**

### CODE REVIEW

File: clients/python/src/tensorzero/client.py [hunk 1]
```diff
@@ -493,23 +487,30 @@ async def __aexit__(
...
-                data = line[6:].strip()
-                if data == "[DONE]":
-                    break
-                try:
-                    parsed_data: Dict[str, Any] = json.loads(data)
-                    yield parse_inference_chunk(parsed_data)
-                except json.JSONDecodeError:
-                    self.logger.error(f"Failed to parse SSE data: {data}")
-        await response.aclose()
+        async with self.client.stream("POST", url, json=data) as response:
```
Comment (virajmehta) [inline clients/python/src/tensorzero/client.py:500]: can you add documentation of the weird httpx behavior here?

File: clients/python/src/tensorzero/client.py [hunk 2]
```diff
@@ -327,26 +321,33 @@ def __exit__(
...
-                    self.logger.error(f"Failed to parse SSE data: {data}")
-        response.close()
+        with self.client.stream("POST", url, json=data) as response:
+            try:
+                response.raise_for_status()
+            except httpx.HTTPStatusError as e:
+                response.read()
+                raise TensorZeroError(response) from e
+
+            for line in response.iter_lines():
```
Comment (virajmehta) [inline clients/python/src/tensorzero/client.py:341]: can you add a comment documenting weird httpx streaming behavior here?

File: clients/python/tests/test_client.py [hunk 3]
```diff
@@ -85,14 +84,18 @@ async def test_async_inference_streaming(async_client):
...
+        previous_chunk_timestamp = time()
         chunks.append(chunk)
-        if first_chunk_duration is None:
-            first_chunk_duration = time() - start_time
-    last_chunk_duration = time() - start_time - first_chunk_duration
-    assert last_chunk_duration > first_chunk_duration + 0.1
+
+    assert last_chunk_duration > 0.01
+
     expected_text = [
```
Comment (virajmehta) [inline clients/python/tests/test_client.py:99]: am I correct in that we were already testing this then?

File: clients/python/tests/test_client.py [hunk 4]
```diff
@@ -546,17 +559,42 @@ def test_sync_inference_streaming(sync_client):
...
             input={
                 "system": {"assistant_name": "Alfred Pennyworth"},
                 "messages": [{"role": "user", "content": "Hello"}],
             },
             stream=True,
         )
+
+        for chunk in stream:
+            pass
+
```
Comment (virajmehta) [inline clients/python/tests/test_client.py:573]: comment why

**Review by virajmehta — APPROVED** [review]
small questions and doc fixes otherwise LGTM

File: clients/python/tests/test_client.py (hunk 3 above)
Comment (GabrielBianconi) [reply clients/python/tests/test_client.py:99]: yeah, but somewhat incorrectly (we were checking the duration of all chunks after the first combined)

**Pull request merged**